io/include/pcl/io/impl/pcd_io.hpp

```diff
@@ -494,60 +494,42 @@ pcl::PCDWriter::writeASCII (const std::string &file_name, const pcl::PointCloud<
           {
             int8_t value;
             memcpy (&value, reinterpret_cast<const char*> (&cloud.points[i]) + fields[d].offset + c * sizeof (int8_t), sizeof (int8_t));
-            if (std::isnan (value))
-              stream << "nan";
-            else
-              stream << boost::numeric_cast<uint32_t>(value);
+            stream << boost::numeric_cast<uint32_t>(value);
```

Comment:
I think there's a bug here, not introduced by you. The cast should be to an a signed `int` type.

Reply:
`uint32_t` is an `unsigned int`, and Values of a `uint8_t` fits always into a `uint32_t`. So where is the bug?

Reply:
We are in `int8` branch here (signed).

Reply:
Ups, was 5 lines below. Will fix it.